glean_parser/metrics.py

```diff
@@ -312,7 +312,16 @@ def allowed_extra_keys(self):
     def allowed_extra_keys_with_types(self):
         # Sort keys so that output is deterministic
         return sorted(
-            [(k, v["type"]) for (k, v) in self.extra_keys.items()], key=lambda x: x[0]
+            [(k, v["type"]) for (k, v) in self.extra_keys.items() if "type" in v],
```

Comment:
Hm, I didn't consider the case where only some of the extra keys have a type set. Edge case, but possible I guess.

We can simplify it though and set a default type here as a fallback:

```suggestion
            [(k, v.get("type", "string")) for (k, v) in self.extra_keys.items() if "type" in v],
```

Then you don't need the `_without_types` method below and the template won't need the extra call either.

Reply:
(plus this makes it work for the other languages too)